### PR TITLE
feat: dynamically react to language change

### LINKED ISSUE #1540: feat(i18n): translation strings inside components should be reactive
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

## Steps to reproduce the issue

1. set a diffrent language without page refresh -> global i18n strings inside lyne components will not switch to the current language
2. refresh the page -> now it will be displayed correctly

### PR DESCRIPTION
**Pull request by jeripeierSBB**
Closes #1540

### CODE REVIEW

File: CODING_STANDARDS.md [hunk 1]
```diff
@@ -94,6 +94,23 @@ class ActionElement implements LinkButtonProperties {
 }
 ```
 
+#### I18N
+
+If using strings in components they have to be provided in English, French, German and Italian.
```
Comment (jeripeierSBB) [inline CODING_STANDARDS.md:99]:
```suggestion
If using texts in components they have to be provided in English, French, German and Italian.
```

File: CODING_STANDARDS.md [hunk 2]
```diff
@@ -94,6 +94,23 @@ class ActionElement implements LinkButtonProperties {
...
+As the language can be changed dynamically, you have to listen to the `sbbLanguageChange`
+event and re-render the view. This can be done by marking the field with `@State` (see code below).
+
+```ts
+export class Component {
+  @State() private _lang = documentLanguage();
```
Comment (jeripeierSBB) [inline CODING_STANDARDS.md:105]:
```suggestion
  @State() private _currentLanguage = documentLanguage();
```

File: CODING_STANDARDS.md [hunk 3]
```diff
@@ -94,6 +94,23 @@ class ActionElement implements LinkButtonProperties {
...
+
+  @Listen('sbbLanguageChange', { target: 'document' })
+  public handleLanguageChange(event: SbbLanguageChangeEvent): void {
+    this._lang = event.detail;
```
Comment (jeripeierSBB) [inline CODING_STANDARDS.md:109]:
```suggestion
    this._currentLanguage = event.detail;
```

File: src/global/interfaces/link-button-properties.ts [hunk 4]
```diff
@@ -109,10 +109,12 @@ export function getLinkButtonBaseAttributeList(
 /**
  * Lists all attributes for a link; undefined/null properties are not set.
  * @param linkProperties link properties
- * @param buttonProperties (optional) In the case of a mixed button and anchor variant, pass in also button properties, which enables the possibility of a disabled link
+ * @param currentLanguage language for accessibility strings.
```
Comment (jeripeierSBB) [inline src/global/interfaces/link-button-properties.ts:112]:
```suggestion
 * @param currentLanguage language for accessibility texts.
```

File: src/global/interfaces/link-button-properties.ts [hunk 5]
```diff
@@ -161,15 +163,17 @@ export function getButtonAttributeList(buttonProperties: ButtonProperties): Reco
 /**
  * Set default render variables for link case.
  * @param linkProperties used to set the 'attributes' property.
- * @param buttonProperties (optional) In the case of a mixed button and anchor variant, pass in also button properties, which enables the possibility of a disabled link
+ * @param currentLanguage language for accessibility strings.
```
Comment (jeripeierSBB) [inline src/global/interfaces/link-button-properties.ts:166]:
```suggestion
 * @param currentLanguage language for accessibility texts.
```

File: src/global/interfaces/link-button-properties.ts [hunk 6]
```diff
@@ -204,29 +208,33 @@ export function getLinkButtonStaticRenderVariables(
 /**
  * Set default render variables based on the 'default' condition, checking first `isStatic` parameter, then the `href`.
  * @param linkButtonProperties used to set the 'attributes' property and to check for `href` value.
+ * @param currentLanguage language for accessibility strings.
```
Comment (jeripeierSBB) [inline src/global/interfaces/link-button-properties.ts:211]:
```suggestion
 * @param currentLanguage language for accessibility texts.
```

File: src/global/interfaces/link-button-properties.ts [hunk 7]
```diff
@@ -204,29 +208,33 @@ export function getLinkButtonStaticRenderVariables(
...
-    return getLinkRenderVariables(linkButtonProperties, linkButtonProperties);
+    return getLinkRenderVariables(linkButtonProperties, currentLanguage, linkButtonProperties);
   }
   return getButtonRenderVariables(linkButtonProperties);
 }
 
 /**
  * Returns the link render variables or static variables if there is no href property.
  * @param linkProperties used to set variables and to check if href property is set.
+ * @param currentLanguage language for accessibility strings.
```
Comment (jeripeierSBB) [inline src/global/interfaces/link-button-properties.ts:230]:
```suggestion
 * @param currentLanguage language for accessibility texts.
```

**Review by kyubisation — APPROVED** [review]
LGTM 👍

**Pull request merged**